src/main/java/com/cyber/mysticism/tarot/repository/TarotDeckRepository.java

```diff
@@ -0,0 +1,50 @@
+package com.cyber.mysticism.tarot.repository;
+
+import com.cyber.mysticism.tarot.json.Card;
+import com.cyber.mysticism.tarot.json.Deck;
+import com.cyber.mysticism.tarot.repository.exceptions.TarotDeckAccessException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Repository;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+@Repository
+public class TarotDeckRepository {
+
+    private final List<Card> cards;
+    private final ObjectMapper objectMapper;
+    private final Resource resourceFile;
+
+    @Autowired
+    public TarotDeckRepository() throws TarotDeckAccessException {
+        ResourceLoader resourceLoader = new DefaultResourceLoader();
+        this.resourceFile = resourceLoader.getResource("classpath:static/tarot-deck/tarot-cards.json");
+        this.objectMapper = new ObjectMapper();
+        this.cards = parseCards();
+    }
+
+    public List<Card> getCards() {
+        return cards;
+    }
+
+    private List<Card> parseCards() throws TarotDeckAccessException {
+        Deck deck = parseDeck();
+        return deck.cards();
+    }
+
+    private Deck parseDeck() throws TarotDeckAccessException {
+        try (BufferedReader reader = Files.newBufferedReader(Path.of(resourceFile.getURI()))) {
+            return objectMapper.readValue(reader, Deck.class);
+        } catch (IOException e) {
```

Comment:
**Suggestion:** Consider using try-with-resources for BufferedReader to ensure that the resource is closed at the end of the statement. [best practice]
```suggestion
        try (BufferedReader reader = Files.newBufferedReader(Path.of(resourceFile.getURI()))) {
            return objectMapper.readValue(reader, Deck.class);
        } catch (IOException e) {
            throw new TarotDeckAccessException("An error occurred while accessing data", e);
        } finally {
            if (reader != null) {
                reader.close();
            }
        }
```